Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,383 @@
+/*global define*/
+define([
+        './clone',
+        './Check',
+        './defined',
+        './defineProperties',
+        './Heap',
+        './isBlobUri',
+        './isDataUri',
+        './RequestState',
+        '../ThirdParty/Uri',
+        '../ThirdParty/when'
+    ], function(
+        clone,
+        Check,
+        defined,
+        defineProperties,
+        Heap,
+        isBlobUri,
+        isDataUri,
+        RequestState,
+        Uri,
+        when) {
+    'use strict';
+
+    /**
+     * Tracks the number of active requests and prioritizes incoming requests.
+     *
+     * @exports RequestScheduler
+     *
+     * @private
+     */
+    function RequestScheduler() {
+    }
+
+    /**
+     * The maximum number of simultaneous active requests. Un-throttled requests do not observe this limit.
+     * @type {Number}
+     * @default 50
+     */
+    RequestScheduler.maximumRequests = 50;
+
+    /**
+     * The maximum number of simultaneous active requests per server. Un-throttled requests do not observe this limit.
+     * @type {Number}
+     * @default 50
+     */
+    RequestScheduler.maximumRequestsPerServer = 6;
+
+    /**
+     * The maximum size of the priority heap. This limits the number of requests that are sorted by priority. Only applies to requests that are not yet active.
+     * @type {Number}
+     * @default 20
+     */
+    RequestScheduler.priorityHeapSize = 20;
+
+    /**
+     * Specifies if the request scheduler should throttle incoming requests, or let the browser queue requests under its control.
+     * @type {Boolean}
+     * @default true
+     */
+    RequestScheduler.debugThrottle = true;
+
+    /**
+     * When true, log statistics to the console every frame
+     * @type {Boolean}
+     * @default false
+     */
+    RequestScheduler.debugShowStatistics = false;
+
+    defineProperties(RequestScheduler, {
+        /**
+         * Returns the statistics used by the request scheduler.
+         *
+         * @memberof RequestScheduler
+         *
+         * @type Object
+         * @readonly
+         */
+        statistics : {
+            get : function() {
+                return statistics;
+            }
+        }
+    });
+
+    function sortRequests(a, b) {
```

Comment:
Normally, we would put everything from here until 107 at the top of the file.  Kind of weird having it in the middle like this (especially since statistics is used above.